src/shared/Specimen.ts

```diff
@@ -0,0 +1,100 @@
+import type {VisualizerInterface} from '../visualizers/VisualizerInterface'
+import type {SequenceInterface} from '../sequences/SequenceInterface'
+
+/**
+ * This class represents a specimen, containing a visualizer,
+ * a sequence, and the set of all parameters that make both of them up.
+ * Specimens can be converted to and from URLs so that they can be saved.
+ */
+export class Specimen {
+    private visualizer: VisualizerInterface
+    private sequence: SequenceInterface
+    private location?: HTMLElement
+    private isSetup: boolean = false
```

Comment:
Oh, I guess it is legal to re-set-up, and the enforcement is already taken care of by Visualizer.inhabit()? If so, you can mostly ignore this comment, but maybe just mention that there will always be a unique location as ensured by inhabit()?  But that does bring up another concern: I don't see that the new code is ever calling Visualizer.depart(), which is intended to be called when a Visualizer is being taken out of the DOM. So please see if there is some place where you should be "cleaning up" visualizers that have been setup.

Reply:
Since in this PR there is only ever one visualizer loaded that is loaded indefinitely, such functionality for departing doesn't really exist yet. However, for the PR with switching visualizers, this will definitely be implemented. And also, it is the responsibility of the visualizer to ensure that resources from previous inhabits are cleaned up whenever `inhabit` is called again, according to the docs. So currently, explicitly calling `depart` isn't necessary

Reply:
All fair enough. Resolving.